activerecord/test/cases/migration/schema_definitions_test.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require "cases/helper"
+
+module ActiveRecord
+  class Migration
+    class SchemaDefinitionsTest < ActiveRecord::TestCase
+      attr_reader :connection
+
+      def setup
+        @connection = ActiveRecord::Base.connection
+      end
+
+      def test_build_create_table_definition_with_block
+        td = connection.build_create_table_definition :test do |t|
+          t.column :foo, :string
+        end
+
+        id_column = td.columns.find { |col| col.name == "id" }
+        assert_predicate id_column, :present?
+        assert id_column.type
+        assert id_column.sql_type
```

Comment:
If it's worth it, I can add adapter-specific tests and be explicit with these assertions on the types / SQL types